website/pages/blog/type-safe-your-api.mdx

```diff
@@ -0,0 +1,132 @@
+```typescript
+async function main() {
+  const topTracks = await getTopTracks();
+  console.log(topTracks);
+}
+
+main();
+```
+
+While this approach works, it has some drawbacks:
+
+- The `fetchWebApi` function lacks type safety, making it uncertain what data will return from the API.
+- You need to specify the parameters of the `fetchWebApi` function manually.
+- The `fetchWebApi` function allows calling any endpoint, even if it's not supported by the API.
+- The `topTracks` variable lacks type safety according to the API documentation.
+
+## How I Use Web API with feTS?
+To use the Spotify API with feTS, follow these steps:
+
+1. Create a new TypeScript project and install feTS from [here](https://the-guild.dev/openapi/fets/client/quick-start).
+
+2. Create a TypeScript file that exports your OpenAPI document as a constant:
+```typescript filename="openapi.ts"
+export default { openapi: '3.0.0' /* ... */ } as const;
+```
+In our case, we will use the Spotify API, so we can retrieve the OpenAPI document from [here](https://developer.spotify.com/reference/web-api/open-api-schema.yaml). Since this OpenAPI document is in YAML format, we need to convert it to JSON format before using it in our project. You can find an example of how to perform the conversion [here](https://github.com/TuvalSimha/feTS-example/blob/main/convert-yaml-to-json.ts).
+
+3. Now, we are ready to use feTS to create a type-safe API client:
+```typescript filename="spotify.ts"
+import { createClient, Mutable } from "fets";
+import spotifyOpenapi from "./spotifyOpenapi";
+
+const client = createClient<Mutable<typeof spotifyOpenapi>>({
+  endpoint: 'https://api.spotify.com/v1',
+});
+```
+
+We import the `createClient` and `Mutable` functions from the feTS package and use the `createClient` function to instantiate a new client. Furthermore, we import the `spotifyOpenapi` file we created in the previous step and use it as a generic type for the `createClient` function.
+
+4. Now, to retrieve our top 5 tracks from the last 30 days using the Spotify API, we can do the following:
+```typescript filename="spotify.ts"
+const topTracks = await client["/me/top/{type}"].get({
+  params: {
+    type: "tracks",
+  },
+  query: {
+    time_range: "short_term",
+    limit: 5,
+  },
+});
+```
+By using the client in this manner, we gain type safety for the endpoint, parameters, and query.
+
+![endpoint](/blog-assets/type-safe-your-api/endpoint.png)
+
+
+5. In conclusion, we know the type of the `topTracks` variable out-of-the-box.
+```typescript filename="spotify.ts"
+const topTracksResponse = await topTracks.json();
+console.log(topTracksResponse);
+```
+
+## Advantages and Disadvantages of Using feTS for Spotify API Integration
+
+### Advantages of feTS
+- Robust Type Safety: feTS eliminates the need for manual type definitions or code generation by inferring types from the OpenAPI specification. This ensures reliable and type-safe API interactions.
```

Comment:
I would maybe even specifically name some clients that don't have type-safety as an examples (axios and maybe others from the announcement blog post), and also Swagger Codegen as an example of codegen one